packages/editor-sdk/src/components/Widgets/StyleWidgets/SpaceWidget/SpaceEditPanel.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  Button,
+  Popover,
+  PopoverContent,
+  chakra,
+  VStack,
+  Input,
+  HStack,
+  PopoverTrigger,
+  InputGroup,
+  InputRightAddon,
+} from '@chakra-ui/react';
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+import { isNaN } from 'lodash-es';
+import { SpaceValue } from './type';
+
+const shortcutsStyle = css`
+  column-gap: 8px;
+  row-gap: 6px;
+  button {
+    margin: 0 !important;
+  }
+`;
+
+export type SpaceConfig = {
+  hasAuto: boolean;
+  shortcuts: number[];
+};
+
+type SpaceEditPanelProps = {
+  value: SpaceValue;
+  onChange: (value: SpaceValue) => void;
+  spaceConfig: SpaceConfig;
+};
+
+export const SpaceEditPanel: React.FC<SpaceEditPanelProps> = ({
+  value,
+  onChange,
+  children,
+  spaceConfig,
+}) => {
+  const [inputValue, setInputValue] = useState<SpaceValue>(value);
+  const { hasAuto, shortcuts } = spaceConfig;
+
+  useEffect(() => {
+    setInputValue(value);
+  }, [value]);
+
+  return (
+    <Popover closeOnBlur placement="bottom">
+      <PopoverTrigger>{children}</PopoverTrigger>
+      <chakra.div
+        sx={{
+          '.chakra-popover__popper': {
+            inset: '0px auto auto -3px !import',
+          },
+        }}
+      >
+        <PopoverContent
+          mt="1"
+          p="2"
+          opacity="0"
+          rounded="md"
+          maxH="350px"
+          shadow="base"
+          zIndex="popover"
+          overflowY="auto"
+          width="300px"
+          _focus={{ boxShadow: 'none' }}
+        >
+          <VStack w="full">
+            <HStack w="full">
+              <InputGroup>
+                <Input
```

Comment:
We should use `ExpressionEditor` here.

Reply:
Using the `ExpressionWidget` instead, because it could display the expression value and error.

Reply:
There are two considerations for not using `ExpressionWidget` here.
1. there seem to be very few scenarios where margin or padding is needed in association with other elements, so the implementation allows only the specified values to be entered. Illegal input, such as 'something wrong', is silently corrected to the previous value
2. the original intention of the style widget was to simply set some common css properties, so there may not be much need for expressions? And we can still use complex expressions via the css editor.

Reply:
The `ExpressionWidget` is still necessary because it is a widget which means so it could be used in any other where, not just style trait form and there is no css editor in other place.